lib/ansible/modules/pip.py

```diff
@@ -611,6 +613,61 @@ def setup_virtualenv(module, env, chdir, out, err):
     return out, err, cmd
 
 
+def _normalize_vcs_packages(module: AnsibleModule,
+                            package_list: list[Package],
+                            pip: list[str]) -> list[Package]:
+    """Return list[Package] where all vcs URL packages are converted to actual package names"""
+    vcs_packages = [str(pkg) for pkg in package_list if _is_vcs_url(str(pkg))]
+
+    if not vcs_packages:
+        return package_list
+
+    other_packages = [pkg for pkg in package_list if not _is_vcs_url(str(pkg))]
+
+    # First, install as a dry-run report to get JSON output
+    frozen_env = os.environ.copy()
+    os.environ['TTY_COMPATIBLE'] = "0"
+    if os.environ.get('FORCE_COLOR'):
```

Comment:
FYI `dict.get()` isn't a universally reliable way to check if there's a key in a dict. `if 'FORCE_COLOR' in os.environ:` would be more accurate. That, or attempting to `del os.environ['FORCE_COLOR']` and wrapping it `with contextlib.suppress(KeyError):` or a try/except.